File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/permissions/storage/write/request/StorageWritePermissionsRequestedBroadcaster.java
package com.lasthopesoftware.bluewater.client.library.permissions.storage.write.request;

import android.content.Intent;
import android.support.v4.content.LocalBroadcastManager;

import com.lasthopesoftware.bluewater.client.library.permissions.storage.read.request.StorageReadPermissionsRequestedBroadcaster;
import com.lasthopesoftware.bluewater.shared.MagicPropertyBuilder;

/**
 * Created by david on 7/3/16.
 */
public class StorageWritePermissionsRequestedBroadcaster implements IStorageWritePermissionsRequestedBroadcaster {
	public final static String WritePermissionsNeeded = MagicPropertyBuilder.buildMagicPropertyName(StorageReadPermissionsRequestedBroadcaster.class, "WritePermissionsNeeded");
	public final static String WritePermissionsLibraryId = MagicPropertyBuilder.buildMagicPropertyName(StorageWritePermissionsRequestedBroadcaster.class, "WritePermissionsLibraryId");

	private final LocalBroadcastManager localBroadcastManager;

	public StorageWritePermissionsRequestedBroadcaster(LocalBroadcastManager localBroadcastManager) {
		this.localBroadcastManager = localBroadcastManager;
	}

	@Override
	public void sendWritePermissionsNeededBroadcast(int libraryId) {
		final Intent writePermissionsNeededIntent = new Intent(WritePermissionsNeeded);
		writePermissionsNeededIntent.putExtra(WritePermissionsLibraryId, libraryId);
		localBroadcastManager.sendBroadcast(writePermissionsNeededIntent);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/request/write/StorageWritePermissionsRequestedBroadcaster.kt
package com.lasthopesoftware.bluewater.client.browsing.library.request.write

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.shared.messages.application.ApplicationMessage
import com.lasthopesoftware.bluewater.shared.messages.application.SendApplicationMessages

class StorageWritePermissionsRequestedBroadcaster(private val sendApplicationMessages: SendApplicationMessages) :
	BroadcastWritePermissionsRequest {

	override fun sendWritePermissionsNeededBroadcast(libraryId: LibraryId) =
		sendApplicationMessages.sendMessage(WritePermissionsNeeded(libraryId))

	class WritePermissionsNeeded(val libraryId: LibraryId) : ApplicationMessage
}
